subs/pantry/src/Pantry/Types.hs

```diff
@@ -438,9 +442,27 @@ data PackageTarball = PackageTarball
   }
   deriving Show
 
--- | Parse Cabal types using 'Distribution.Text.Text'.
-parseC :: Distribution.Text.Text a => String -> Maybe a
-parseC = Distribution.Text.simpleParse
+-- | Modified version of Cabal's parser - we don't need null version
+-- in package indentifiers
+parsePackageIdentifier :: String -> Maybe PackageIdentifier
+parsePackageIdentifier str =
+    case [p | (p, s) <- Parse.readP_to_S parser str, all isSpace s] of
+        [] -> Nothing
+        (p:_) -> Just p
+  where
+    parser = do
```

Comment:
Looks like a better workaround for the Cabal bug. Can you add a comment explaining why we have to go through these hoops so someone in the future doesn't "simplify" this code?

Reply:
Added more detailed comments